examples/puzzles/river/river.qnt

```diff
@@ -0,0 +1,112 @@
+// -*- mode: Bluespec; -*-
+/**
+ * This is an example of a river-crossing puzzle [1].
+ * The specification itself is quite simple, so it can serve as a good
+ * starting example.
+ *
+ * [1]: https://en.wikipedia.org/wiki/River_crossing_puzzle
+ *
+ * "The fox, goose and bag of beans puzzle, in which a farmer must transport a
+ * fox, goose and bag of beans from one side of a river to another using a boat
```

Comment:
You might want to say below this paragraph that you're actually using different items.

I was confused by the sudden appearance of a 🐐

Reply:
Yeah, I am using the last one. Will add a sentence.